multisig-wallet/project/contracts/multisig-contract/tests/utils/setup.rs

```diff
@@ -227,40 +226,40 @@ pub(crate) fn call_parameters(
 ) -> Transaction {
     match with_value {
         false => Transaction {
-            contract_call_params: Some(ContractCallParams {
+            contract_identifier: deployer.contract.contract_id().try_into().unwrap(),
```

Comment:
Looking at this function it creates a struct that is almost identical in both cases. 
I suggest creating a default struct that can be mutated and then match on the `with_value`.
It looks like 2 values are changed between the two arms so you can return those into a tuple and then mutate the struct. This way you're not duplicating a lot of code which also increases the cognitive load, reduces DRY-ness.

Reply:
I've done something similar, but have had to account for the fact that the fields that change are in the `transaction_parameters` field, which takes a variant of the enum `TransactionParameters`. Still DRY-er though:
```rust
if with_value {
        transaction.transaction_parameters = TransactionParameters::Call(ContractCallParams {
            function_selector: Bytes(fn_selector!(update_deposit(Address, u64))),
            transfer_params: TransferParams {
                asset_id: base_asset_contract_id(),
                value: Some(DEFAULT_TRANSFER_AMOUNT),
            },
            ..contract_call_params
        });
    }
```